sdk/storage/Azure.Storage.Blobs.AspNetCore.DataProtection/src/AzureStorageBlobDataProtectionBuilderExtensions.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.Extensions.DependencyInjection;
+
+#pragma warning disable AZC0001 // Extension methods have to be in the correct namespace to appear in intellisense.
+namespace Microsoft.AspNetCore.DataProtection
+#pragma warning disable
+{
+    /// <summary>
+    /// Contains Azure-specific extension methods for modifying a
+    /// <see cref="IDataProtectionBuilder"/>.
+    /// </summary>
+    public static class AzureStorageBlobDataProtectionBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the data protection system to persist keys to the specified path
+        /// in Azure Blob Storage.
+        /// </summary>
+        /// <param name="builder">The builder instance to modify.</param>
+        /// <param name="blobUri">The full URI where the key file should be stored.
+        /// The URI must contain the SAS token as a query string parameter.</param>
+        /// <returns>The value <paramref name="builder"/>.</returns>
+        /// <remarks>
+        /// The container referenced by <paramref name="blobUri"/> must already exist.
+        /// </remarks>
+        public static IDataProtectionBuilder PersistKeysToAzureBlobStorage(this IDataProtectionBuilder builder, Uri blobUri)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+            if (blobUri == null)
+            {
+                throw new ArgumentNullException(nameof(blobUri));
+            }
+
+            var uriBuilder = new BlobUriBuilder(blobUri);
+            BlobClient client;
+
+            // The SAS token is present in the query string.
+            if (uriBuilder.Sas == null)
+            {
+                client = new BlobClient(blobUri, new DefaultAzureCredential());
```

Comment:
I guess there's no meaningful case where this would ever be a public access blob, right?  That's the only other case where the Sas would be null with a shared key.

Reply:
Would it actually be nicer to throw if there's no Sas and make users supply the `DefaultAzureCredential` themselves in the other overload that takes a token credential?  I don't have a good feel for how much magic is the right amount of magic in the ASP.NET world.